### PR TITLE
[IMP]sale: Add different search views for different list views.

### PR DESCRIPTION
**Pull request by kme-odoo**
Task: https://www.odoo.com/web#id=21095&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.HrWV5ITEQLXMT2vh

### CODE REVIEW

**Comment by xmo-odoo** [comment]
Since the views are almost identical, why not create a base view with the common elements and then use `primary` view inheritance to create the various views actually needed for the different actions?

**Comment by kme-odoo** [comment]
@xmo-odoo as per your suggestion, we can create common view and inherit that view by creating other search views for each action(If I am getting you correctly), but note that if I am having two view inherting same common view and if I call particular view using search_view_id, then both view will be applied in common view when ir.ui.view applies inheritance.

say for example: I am having common view: view_search_sale_order_common and having two search view for two actions: view_search_common_inherit1 and view_search_common_inherit2, now when fields_view_get is called, at that time get_inheriting_views_arch will return all view which having inherit_id=view_search_sale_order_common, and applies arch of both view(may result in view overridden, last priority view will be applied at the end)

**Comment by xmo-odoo** [comment]
> note that if I am having two view inherting same common view and if I call particular view using search_view_id, then both view will be applied in common view when ir.ui.view applies inheritance.

Not if the children views are marked as primary. "extension" views (the default views with an `inherit_id`) are implicitly applied to their parent, "primary" views (the default for views with no `inherit_id`) are applied to their base only when requested explicitly, and siblings (other "primary" views inheriting the same base) are not applied.

**Comment by kme-odoo** [comment]
@xmo-odoo :+1:

File: addons/sale/sale_view.xml [hunk 1]
```diff
@@ -375,15 +370,69 @@
             </field>
         </record>
 
+        <record id="view_quotation_inherit_filter" model="ir.ui.view">
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: now give view id name based on new guidelines.

File: addons/sale/sale_view.xml [hunk 2]
```diff
@@ -375,15 +370,69 @@
...
+            <field name="name">sale.order.list.select.quotation.inherit</field>
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: give name something like : sale.order.inherit.quotation

File: addons/sale/sale_view.xml [hunk 3]
```diff
@@ -375,15 +370,69 @@
...
+            <field name="model">sale.order</field>
+            <field name="mode">primary</field>
+            <field name="inherit_id" ref="sale.view_sales_order_filter"/>
+            <field name="arch" type="xml">
+                <xpath expr="//div[@name='filter_box']" position="replace">
+                        <filter string="Quotations" name="draft" domain="[('state','in',('draft','sent'))]" help="Sales Order that haven't yet been confirmed"/>
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: give proper indentation....

File: addons/sale/sale_view.xml [hunk 4]
```diff
@@ -375,15 +370,69 @@
...
+            <field name="model">sale.order</field>
+            <field name="mode">primary</field>
+            <field name="inherit_id" ref="sale.view_sales_order_filter"/>
+            <field name="arch" type="xml">
+                <xpath expr="//div[@name='filter_box']" position="replace">
+                        <filter string="Sales" name="sales" domain="[('state','in',('progress','done'))]" />
+                        <separator/>
+                        <filter string="To Invoice" domain="[('invoice_status','=','to invoice')]" />
+                        <filter string="Upselling" domain="[('invoice_status','=','upselling')]" />
+                        <separator/>
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: give proper indentation . same as above.

File: addons/sale/sale_view.xml [hunk 5]
```diff
@@ -375,15 +370,69 @@
...
+            <field name="arch" type="xml">
+                <xpath expr="//div[@name='filter_box']" position="replace">
+                        <filter string="To Invoice" domain="[('invoice_status','=','to invoice')]" />
+                        <separator/>
+                </xpath>
+            </field>
+        </record>
+
+        <record id="view_upselling_inherit_filter" model="ir.ui.view">
+            <field name="name">sale.order.list.select.upselling.inherit</field>
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: name should be like : sale.order.filter.inherit.upselling

File: addons/sale/sale_view.xml [hunk 6]
```diff
@@ -434,11 +485,10 @@
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">sale.order</field>
             <field name="view_type">form</field>
-            <field name="view_id" ref="view_quotation_tree"/>
+            <field name="view_id" ref="view_sales_order_filter"/>
```
Comment (atp-odoo) [inline addons/sale/sale_view.xml]: why this change? in view_id you changed tree view reference to search view. wrong. correct it. keep as it is.
